src/amo/components/AddonReviewCard/index.js

```diff
@@ -278,40 +281,55 @@ export class AddonReviewCardBase extends React.Component<InternalProps> {
       _config,
       addon,
       className,
+      clientApp,
       deletingReview,
       editingReview,
       errorHandler,
       flaggable,
       i18n,
-      siteUserHasReplyPerm,
+      lang,
       replyingToReview,
       review,
       shortByLine,
       showRating,
       siteUser,
+      siteUserHasReplyPerm,
       verticalButtons,
     } = this.props;
 
     let byLine;
+    const noAuthor = shortByLine || this.isReply();
 
     if (review) {
       const timestamp = i18n.moment(review.created).fromNow();
-      if (shortByLine || this.isReply()) {
-        // translators: Example in English: "posted last week"
-        byLine = i18n.sprintf(i18n.gettext('posted %(timestamp)s'), {
-          timestamp,
-        });
-      } else {
-        byLine = (
-          <span className="AddonReviewCard-authorByLine">
-            {/* translators: Example in English: "by UserName123, last week" */}
-            {i18n.sprintf(i18n.gettext('by %(authorName)s, %(timestamp)s'), {
+      const linkStart = oneLineTrim`<a href="/${lang}/${clientApp}/addon/${
+        review.addonSlug
+      }/reviews/${review.id}/">`;
+      const byLineString = noAuthor
+        ? // translators: Example in English: "posted last week"
+          i18n.gettext('posted %(linkStart)s%(timestamp)s%(linkEnd)s')
+        : // translators: Example in English: "by UserName123, last week"
+          i18n.gettext(
+            'by %(authorName)s, %(linkStart)s%(timestamp)s%(linkEnd)s',
+          );
+
+      byLine = (
+        <span
+          className={makeClassName('', {
+            'AddonReviewCard-authorByLine': !noAuthor,
+          })}
+          // eslint-disable-next-line react/no-danger
+          dangerouslySetInnerHTML={sanitizeHTML(
+            i18n.sprintf(byLineString, {
```

Comment:
Can we make these links underlined but `$grey-50` like the text? I think that would make the header look more cohesive. What do you think?

<img width="841" alt="screenshot 2018-09-18 09 39 12" src="https://user-images.githubusercontent.com/55398/45695379-e6cac580-bb26-11e8-8bfd-0b8d234d4788.png">

Reply:
Yes, sorry, I recall you mentioned that earlier, but it got lost in the comments. Tbh I'm experiencing fatigue with this PR, but it seems like it's getting very close.

Reply:
It is super close, yes! It was a huge feature. For future reference, it would have been easier to review if you'd split the actions and reducer code into a separate patch.

As a general rule, I think 400 lines of changes is optimal for a review. Pushing it to 600 is usually OK but anything beyond that just becomes difficult.

Reply:
My apologies. I think when I started it I didn't realize how big it was going to be, so I just started plowing through everything. I will try to take some time to think about chunking the work in the future before I just jump in and start coding.